### PR TITLE
Connects to AgileVentures/MetPlus_tracker#133

### PR DESCRIPTION
**Pull request by patmbolger**
also,

Connects to AgileVentures/MetPlus_tracker#186

### CODE REVIEW

**Comment by patmbolger** [comment]
This PR is the first phase of event management.

In this phase, events result in real-time popups.  The events are 1) job seeker joins PETS, 2) company registers request to be a partner with agency.  Any agency person logged in at the time will see a popup for each event.

Second phase will add emails sent for each event as well.  This may be via simple blocking code.  If so, I will defer async (background) email sending to a 3rd phase.

File: app/controllers/job_seekers_controller.rb [hunk 1]
```diff
@@ -25,11 +25,11 @@ def update
     @jobseeker = JobSeeker.find(params[:id])
 
     person_params = jobseeker_params
-    if person_params['password'].to_s.length == 0
+    if person_params['password'].to_s.length == 0
```
Comment (joaopapereira) [inline app/controllers/job_seekers_controller.rb]: Why change this?

File: app/controllers/job_seekers_controller.rb [hunk 2]
```diff
@@ -25,11 +25,11 @@ def update
     @jobseeker = JobSeeker.find(params[:id])
 
     person_params = jobseeker_params
-    if person_params['password'].to_s.length == 0
+    if person_params['password'].to_s.length == 0 
        person_params.delete('password')
        person_params.delete('password_confirmation')
     end
-
+
```
Comment (joaopapereira) [inline app/controllers/job_seekers_controller.rb]: Why change this?

File: app/controllers/job_seekers_controller.rb [hunk 3]
```diff
@@ -39,7 +39,7 @@ def update
        render 'edit'
     end
   end
-
+
```
Comment (joaopapereira) [inline app/controllers/job_seekers_controller.rb]: Why change this?

File: app/models/event.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class Event
+  include ActiveModel::Model
+
+  def self.create(evt_type, evt_data)
+    case evt_type
+    when :JS_REGISTER     # job seeker registered
+      Pusher.trigger('pusher_control', 'js_registered', evt_data)
```
Comment (joaopapereira) [inline app/models/event.rb]: Maybe we should put these string into static class variables so it is easier to change if we need

File: config/routes.rb [hunk 5]
```diff
@@ -1,14 +1,14 @@
 Rails.application.routes.draw do
 
-  devise_for :users,
-      :path_names => {sign_up: 'new', sign_out: 'logout',
-                      sign_in: 'login' },
-      :controllers => {invitations: 'people_invitations',
-                       confirmations: 'users/confirmations' }
+  devise_for :users, :path_names => {:sign_up => "new", :sign_out => 'logout',
```
Comment (joaopapereira) [inline config/routes.rb]: Any particular reason behind the change of formatting?

**Comment by joaopapereira** [comment]
Looks good, please review my comments, and I would prefer that someone else would review the code as well

File: app/controllers/job_seekers_controller.rb (hunk 3 above)
Comment (patmbolger) [inline app/controllers/job_seekers_controller.rb]: I don't know how these changes were introduced in this file.

File: config/routes.rb (hunk 5 above)
Comment (patmbolger) [inline config/routes.rb]: Not really other than I had to add a key-value pair so fiddled a little with alignment at the same time.

File: app/controllers/job_seekers_controller.rb (hunk 3 above)
Comment (joaopapereira) [inline app/controllers/job_seekers_controller.rb]: If you didnt change the file, please revert the change

**Comment by joaopapereira** [comment]
Looks good going to merge the PR
